libbeat/common/mapstr.go

```diff
@@ -380,3 +389,43 @@ func (op putOperation) Do(key string, data MapStr) (interface{}, error) {
 	data[key] = op.Value
 	return existingValue, nil
 }
+
+type setValueOperation struct {
+	Value interface{}
+}
+
+func (op setValueOperation) Do(key string, data MapStr) (interface{}, error) {
+	if op.Value == nil {
+		return data, nil
+	}
+
+	switch op.Value.(type) {
+	case *bool:
+		if newVal := op.Value.(*bool); newVal != nil {
+			data[key] = *newVal
+		}
+	case *int:
+		if newVal := op.Value.(*int); newVal != nil {
+			data[key] = *newVal
+		}
+	case *string:
+		if newVal := op.Value.(*string); newVal != nil {
+			data[key] = *newVal
+		}
+	case MapStr:
```

Comment:
These empty checks can be added to the appropriate `normalizeX` function in [libbeat/common/event.go](https://github.com/elastic/beats/blob/25fdff884e53a89b18207dd50e8c548361bdb851/libbeat/common/event.go).